evm-template/runtime/src/configs/asset_config.rs

```diff
@@ -0,0 +1,185 @@
+    types::{AccountId, AssetId, Balance},
+    AssetManager, Assets, Balances, Runtime, RuntimeCall, RuntimeEvent, RuntimeOrigin,
+};
+
+parameter_types! {
+    pub const AssetDeposit: Balance = 10 * CENTS;
+    pub const AssetAccountDeposit: Balance = deposit(1, 16);
+    pub const ApprovalDeposit: Balance = MILLICENTS;
+    pub const StringLimit: u32 = 50;
+    pub const MetadataDepositBase: Balance = deposit(1, 68);
+    pub const MetadataDepositPerByte: Balance = deposit(0, 1);
+    pub const RemoveItemsLimit: u32 = 1000;
+}
+
+// Required for runtime benchmarks
+pallet_assets::runtime_benchmarks_enabled! {
+    pub struct BenchmarkHelper;
+    impl<AssetIdParameter> pallet_assets::BenchmarkHelper<AssetIdParameter> for BenchmarkHelper
+    where
+        AssetIdParameter: From<u128>,
+    {
+        fn create_asset_id_parameter(id: u32) -> AssetIdParameter {
+            (id as u128).into()
+        }
+    }
+}
+
+// Foreign assets
+impl pallet_assets::Config for Runtime {
+    type ApprovalDeposit = ApprovalDeposit;
+    type AssetAccountDeposit = AssetAccountDeposit;
+    type AssetDeposit = AssetDeposit;
+    type AssetId = AssetId;
+    type AssetIdParameter = Compact<AssetId>;
+    type Balance = Balance;
+    #[cfg(feature = "runtime-benchmarks")]
+    type BenchmarkHelper = BenchmarkHelper;
+    type CallbackHandle = ();
+    type CreateOrigin = AsEnsureOriginWithArg<EnsureSigned<AccountId>>;
+    type Currency = Balances;
+    type Extra = ();
+    type ForceOrigin = EnsureRoot<AccountId>;
+    type Freezer = ();
+    type MetadataDepositBase = MetadataDepositBase;
+    type MetadataDepositPerByte = MetadataDepositPerByte;
+    type RemoveItemsLimit = RemoveItemsLimit;
+    type RuntimeEvent = RuntimeEvent;
+    type StringLimit = StringLimit;
+    type WeightInfo = pallet_assets::weights::SubstrateWeight<Runtime>; //FIXME: run & update
+}
+
+// Our AssetType. For now we only handle Xcm Assets
+#[derive(Clone, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub enum AssetType {
+    Xcm(xcm::v3::Location),
+}
+impl Default for AssetType {
+    fn default() -> Self {
+        Self::Xcm(xcm::v3::Location::here())
+    }
+}
+
+impl From<xcm::v3::Location> for AssetType {
+    fn from(location: xcm::v3::Location) -> Self {
+        Self::Xcm(location)
+    }
+}
+
+// This can be removed once we fully adopt xcm::v4 everywhere
+impl TryFrom<Location> for AssetType {
+    type Error = ();
+
+    fn try_from(location: Location) -> Result<Self, Self::Error> {
+        Ok(Self::Xcm(location.try_into()?))
+    }
+}
+
+impl From<AssetType> for Option<xcm::v3::Location> {
+    fn from(val: AssetType) -> Self {
+        match val {
+            AssetType::Xcm(location) => Some(location),
+        }
+    }
+}
+
+// Implementation on how to retrieve the AssetId from an AssetType
+// We take it
+impl From<AssetType> for AssetId {
+    fn from(asset: AssetType) -> AssetId {
+        match asset {
```

Comment:
I find `match` statement with a single invariant quite counter intuitive. If you also agree, we can change it to something like `let else` or `if let`. 

We don't need to spent time on this subjective discussions, feel free to leave it as is if you simply disagree :)

Reply:
I disagree.
`if let` will compile if we add a new invariant, `match` will not. This enum will have new invariants for sure.